### PR TITLE
Adapt 1701 cta v2

### PR DESCRIPTION
**Pull request by yvonnetangsu**
# READY FOR REVIEW

# Summary
- This is based on the other CTA branch, but with separate CTA Button and CTA Link component and some shared levers/functions to reduce redundancy.

# Needed By (Date)
- When does this need to be merged by?

# Urgency
- How critical is this PR?

# Steps to Test

1. Do this
1. Then this
2. Then this

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: src/CtaButton/CtaButton.js
Comment (yvonnetangsu) [inline src/CtaButton/CtaButton.js:70]: @sherakama Trying new approach. Is it worth doing this you think?

File: src/common/icon/icon.levers.js
Comment (yvonnetangsu) [inline src/common/icon/icon.levers.js:9]: Shared icon levers

File: src/common/icon/getIconAnimation.js
Comment (yvonnetangsu) [inline src/common/icon/getIconAnimation.js:3]: One of the shared functions that returns a value that is dependent on one of the icon levers..

File: src/CtaButton/CtaButton.js
Comment (yvonnetangsu) [reply src/CtaButton/CtaButton.js:70]: Since we're separating out the CTA Link and Link Button components, we could either duplicate the icon/animation code and have them in each file, or have them shared this way. The duplicate way is more readable, but this way has less redundant code (there might be a better way to structure this).

Comment (yvonnetangsu) [reply src/CtaButton/CtaButton.js:70]: Also, more levers/values are sharable this way, e.g., the button sizes and variants are sharable with the CTA button links.

**Comment by sherakama** [comment]
I'll have a look tomorrow.

**Comment by yvonnetangsu** [comment]
> I'll have a look tomorrow.

Sleep time!

File: src/Button/Button.js
Comment (yvonnetangsu) [inline src/Button/Button.js:50]: Pulling button sizes into common levers/functions as well; I might do that with the variants also.

**Comment by yvonnetangsu** [comment]
@sherakama I know I have one more task left to do for this one - allow custom icon. Let me know if you like this approach of sharing common levers and other code, and I'll finish up.

**Comment by sherakama** [comment]
I'm having a read-through now. I won't block this PR for the added improvement of sending in an icon.

**Comment by yvonnetangsu** [comment]
I think that for the Button component, instead of using the styles from the su-button class from the TW component, I'm just going to do everything in Decanter React with the TW utility classes - much easier dedupe and change, I think. I need to clean that up with the new Dismiss button changes. For the CTA Button, I didn't rely on the su-button class styles, and that works out better I think.

**Review by sherakama — APPROVED** [review]
GTG. 

I think what you have done here is fine. It has been broken up nicely and provides a good developer experience overall. The complexity and abstraction have been tucked away by helper functions that only the curious will need to dive down into. I think the documented options and inspecting the output will be enough for most developers. 

Nice work.

**Comment by sherakama** [comment]
Also, I pushed a couple of changes as well. Please review those before merging.

**Comment by yvonnetangsu** [comment]
> GTG.
> 
> I think what you have done here is fine. It has been broken up nicely and provides a good developer experience overall. The complexity and abstraction have been tucked away by helper functions that only the curious will need to dive down into. I think the documented options and inspecting the output will be enough for most developers.
> 
> Nice work.

Hey thank you for doing the revisions! Yeah, I've looked at the changes and have learned from your review of the Dismiss Button PR so will tackle props and classNames etc in a similar way in the future. Thanks for the new stories too 🚀

**Comment by sherakama** [comment]
Yeah, I am trying to stick to convention with the props where we don't over complicate the simple components and anything that can go directly through to the HTML element as a prop does without us changing the key or messing with the values in between.

**Pull request merged**
